Firestore/Source/Remote/FSTDatastore.mm

```diff
@@ -305,20 +305,19 @@ - (void)invokeRPCWithFactory:(GRPCProtoCall * (^)(void))rpcFactory
   // TODO(mikelehen): We should force a refresh if the previous RPC failed due to an expired token,
   // but I'm not sure how to detect that right now. http://b/32762461
   _credentials->GetToken(
-      /*force_refresh=*/false,
-      [self, rpcFactory, errorHandler](Token result, const int64_t error_code,
-                                       const absl::string_view error_msg) {
-        NSError *error = util::WrapNSError(error_code, error_msg);
+      /*force_refresh=*/false, [self, rpcFactory, errorHandler](util::StatusOr<Token> result) {
+        NSError *error = util::WrapNSError(result.status());
         [self.workerDispatchQueue dispatchAsyncAllowingSameQueue:^{
           if (error) {
```

Comment:
We should defer wrapping the error until we need it for interoperating so that we can get closer to the end state of just using status. This should be:

```c++
if (!result.ok()) {
  errorHandler(MakeNSError(result.status()));
}
// ...
```